src/wp-includes/author-template.php

```diff
@@ -229,13 +229,29 @@ function the_author_meta( $field = '', $user_id = false ) {
  */
 function get_the_author_link() {
 	if ( get_the_author_meta( 'url' ) ) {
-		return sprintf(
+		global $authordata;
+
+		$author_url          = get_the_author_meta( 'url' );
+		$author_display_name = get_the_author();
+
+		$author_url_link_html = sprintf(
 			'<a href="%1$s" title="%2$s" rel="author external">%3$s</a>',
-			esc_url( get_the_author_meta( 'url' ) ),
+			esc_url( $author_url ),
 			/* translators: %s: Author's display name. */
-			esc_attr( sprintf( __( 'Visit %s&#8217;s website' ), get_the_author() ) ),
-			get_the_author()
+			esc_attr( sprintf( __( 'Visit %s&#8217;s website' ), $author_display_name ) ),
+			$author_display_name
 		);
+
+		/**
+		 * Filters the author URL link HTML.
+		 *
+		 * @since 6.0.0
+		 *
+		 * @param string $author_url_link_html    The default rendered author HTML link.
+		 * @param string $author_url              Author's URL.
+		 * @param string $authordata              Author user data.
+		 */
+		return apply_filters( 'author_url_link_html', $author_url_link_html, $author_url, $authordata );
 	} else {
 		return get_the_author();
```

Comment:
What does WP do elsewhere where an item may be an `a href` or a simple string?

There's an argument this is still the link, it's just that the link is unlinked. I hope this makes sense on a reread ;)